### ISSUE TITLE
Python consumer/producer does not relay `traceparent`?

### ISSUE THREAD

**Issue by shift-alt-del** [shift-alt-del/otel-playground#14]
`async-queue` writes to `async-queue-enriched` writes to `T_PV_COUNT`

```
# async-queue
traceparent=00-b425bc3820ecf9cb3f1077f1e908d4c1-87628124d04e136f-01

# async-queue-enriched
traceparent=00-b425bc3820ecf9cb3f1077f1e908d4c1-87628124d04e136f-01,traceparent=00-b1c2dc92b88e3b02a68379e026570c99-8477a161c65c6fd0-01

# T_PV_COUNT
traceparent=00-b1c2dc92b88e3b02a68379e026570c99-62d095c539ab3d09-01
```

To notice, by sending message `producer.send(OUTPUT_TOPIC_NAME, message.value, headers=message.headers)`, `async-queue-enriched` has two different `traceparent`, while `T_PV_COUNT` only relay the latest one, which broke the end-to-end chain.

**Comment by shift-alt-del** [comment]
By adding `produce_hook` and pop the new generated header, seems corrects the end-to-end chain.

Looks like python producer/consumer instrumentor does not work well?

```
def produce_hook(span, args, kwargs):
    kwargs['headers'].pop()
    if span and span.is_recording():
        span.set_attribute("custom_user_attribute_from_produce_hook", "some-value")

KafkaInstrumentor().instrument(produce_hook=produce_hook)
```

<img width="1678" alt="Screen Shot 2022-06-20 at 15 47 16" src="https://user-images.githubusercontent.com/30165935/174542021-64d96633-e5a0-418d-8fbc-e23fa526d7cf.png">
<img width="1250" alt="Screen Shot 2022-06-20 at 15 50 59" src="https://user-images.githubusercontent.com/30165935/174542030-5a1cae52-f35f-4c84-b383-359974dd0fd3.png">

**Comment by shift-alt-del** [comment]
Commands
```
kcat -b broker:9092 -f "%h\n" -t async-queue
kcat -b broker:9092 -f "%h\n" -t async-queue-enriched
kcat -b broker:9092 -f "%h\n" -t T_PV_COUNT
```

**Comment by shift-alt-del** [comment]
TL;DR

- Python Kafka instrument will not switch context for consumer/producer.
- Context needs to be extracted from message header.
- Have to specify parent span by setting previous context.

An example:
```
        ctx = None
        for kk, vv in message.headers:
            if kk == 'traceparent':
                tracing_id = vv.decode('utf-8')
                ctx = TraceContextTextMapPropagator().extract(
                    carrier={'traceparent': tracing_id})
                break

        with tracer.start_as_current_span(name='Otel-MergeContext', context=ctx):
            # todo: enriched data, calling multiple databases here.
            producer.send(OUTPUT_TOPIC_NAME, message.value)
```

<img width="1684" alt="Screen Shot 2022-06-22 at 10 53 22" src="https://user-images.githubusercontent.com/30165935/174926803-1db53a7d-72e2-4ac5-916e-077f2f5ba6c0.png">
